glean-core/android/src/main/java/mozilla/telemetry/glean/net/Upload.kt

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.telemetry.glean.net
+
+import android.util.Log
+import org.json.JSONObject
+import org.json.JSONException
+import com.sun.jna.Structure
+import com.sun.jna.Pointer
+import com.sun.jna.Union
+import mozilla.telemetry.glean.Glean
+import mozilla.telemetry.glean.rust.getRustString
+
+// Rust represents the upload task as an Enum
+// and to go through the FFI that gets transformed into a tagged union.
+// Each variant is represented as an 8-bit unsigned integer.
+const val WAIT = 0
+const val DONE = 2
+const val UPLOAD = 1
+
+@Structure.FieldOrder("tag", "uuid", "path", "body", "headers")
+internal class UploadBody(
+    // NOTE: We need to provide defaults here, so that JNA can create this object.
+    @JvmField val tag: Byte? = null,
```

Comment:
Hm, thinking about this: is it correct to make this an optional/nullable field?
I would expect that it needs to store somewhere if it's null or an actual byte (whereas for pointers it can just pick the null pointer to begin with).

Let's double-check if the documentation says anything about this.